### PR TITLE
epm: add HealthZ interface to get epm service's state by epm client

### PR DESCRIPTION
**Pull request by YangLiang3**
Fixes: #543
Signed-off-by: Liang Yang <liang3.yang@intel.com>

### CODE REVIEW

**Review by hustliyilin — COMMENTED** [review]
please fix the compile check error firstly.

File: epm/pkg/epm/epm.go [hunk 1]
```diff
@@ -122,6 +122,21 @@ func (s *EnclavePoolManagerServer) LoadCache(ctx context.Context, req *v1alpha1.
...
 }
 
+// Healthz represents epm service's running state
+func (s *EnclavePoolManagerServer) Healthz(ctx context.Context, req *v1alpha1.HealthzRequest) (*v1alpha1.HealthzResponse, error) {
+	manager, err := s.getCachePoolManager(req.Type)
+	if err != nil {
+		glog.Errorf("cache pool type %s is not found. error: %++v", req.Type, err)
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	if true != manager.Healthz() {
```
Comment (hustliyilin) [inline epm/pkg/epm/epm.go:132]: change to `if !manager.Healthz()`

File: rune/libenclave/epm/epm.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"os"
 	"path/filepath"
 	"time"
+	"syscall"
```
Comment (hustliyilin) [inline rune/libenclave/epm/epm.go:10]: please use `go fmt` before commiting your codes

File: rune/libenclave/epm/epm.go [hunk 3]
```diff
@@ -46,15 +48,27 @@ func GetCache(ID string, subtype string) *v1alpha1.Enclave {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
+	healthzResp, err := c.Healthz(ctx, &v1alpha1.HealthzRequest{Type: Type})
+	if err != nil {
+		return nil
+	}
+	if healthzResp.Ok != true {
```
Comment (hustliyilin) [inline rune/libenclave/epm/epm.go:55]: ditto

File: rune/libenclave/epm/epm.go [hunk 4]
```diff
@@ -86,6 +101,15 @@ func SaveCache(ID string, subtype string, enclaveinfo *v1alpha1.Enclave) error {
...
```
Comment (hustliyilin) [inline rune/libenclave/epm/epm.go:108]: ditto

**Review by hustliyilin — COMMENTED** [review]
LGTM.

**Review by haosanzi — APPROVED** [review]

File: rune/libenclave/epm/epm.go (hunk 4 above)
Comment (YangLiang3) [reply rune/libenclave/epm/epm.go:108]: ok, thanks. Will pr revision this afternoon.

**Review by hustliyilin — COMMENTED** [review]
please fix the compile check error firstly and execute `go fmt` before committing your codes.

**Review by hustliyilin — APPROVED** [review]

**Review by hustliyilin — APPROVED** [review]
All CI/CD testcases pass and ready to be merged. What's your suggestions? @jiazhang0 @jiazhiguang

**Review by jiazhang0 — APPROVED** [review]

**Review by jiazhiguang — APPROVED** [review]

**Pull request merged**